gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -508,6 +509,91 @@ def create_phylop_grch38_ht() -> hl.Table:
     return ht
 
 
+def get_revel_for_unmatched_transcripts() -> None:
+    """
+    Create Tables with alternative REVEL scores for variants in v4.1 release.
+
+    ..note:
+
+        REVEL was computed using transcripts from Ensembl v64. In the gnomAD v4.0
+        and v4.1 release Tables, transcript information from Ensembl v105 and variant
+        information (locus and alleles combination) were used to ascertain variant
+        REVEL scores for MANE select or canonical transcripts only. This means that
+        variants within 2,414 MANE select transcripts in gnomAD v4.0 and v4.1 are
+        missing REVEL scores because they are not present in Ensembl v64.
+
+        To address this, we annotated the variants within the 2,414 genes with the
+        maximum REVEL score found at the specific locus and allele, rather than the
+        score for the MANE Select transcript.
+
+        The exomes TSV adds REVEL scores to 1,936,321 out of 2,284,296 (87.77%)
+        missense variants within the 2,414 genes. The genomes TSV adds REVEL scores
+        to 528,204 out of 620,799 ( 85.08%) missense variants within the 2,414 genes.
+    """
+
+    def _process_revel():
+        """Process REVEL scores."""
+        revel_csv = "gs://gnomad-insilico/revel/revel-v1.3_all_chromosomes_with_transcript_ids.csv.bgz"
+
+        ht = hl.import_table(
+            revel_csv,
+            delimiter=",",
+            min_partitions=1000,
+            types={"grch38_pos": hl.tstr, "REVEL": hl.tfloat64},
+        )
+        # drop variants that have no position in GRCh38 when lifted over from GRCh37
+        ht = ht.filter(ht.grch38_pos.contains("."), keep=False)
+        ht = ht.select(
+            locus=hl.locus(
+                "chr" + ht.chr, hl.int(ht.grch38_pos), reference_genome="GRCh38"
+            ),
+            alleles=hl.array([ht.ref, ht.alt]),
+            REVEL=ht.REVEL,
+            Transcript_stable_ID=ht.Ensembl_transcriptid.strip().split(";"),
+        )
+        ht = ht.key_by("locus", "alleles")
+        ht = ht.group_by("locus", "alleles").aggregate(REVEL_max=hl.agg.max(ht.REVEL))
+        return ht
+
+    def _filter_release_ht(data_type, genes_list):
+        """Filter release sites to only include genes with missing revel scores."""
+        ht = release_sites(data_type=data_type).ht()
+        ht = process_consequences(ht)
+        ht = filter_vep_transcript_csqs(
```

Comment:
This only filters to canonical at the moment, mane select is false by default.

Reply:
Not sure if the numbers in the note section will need to change because of this

Reply:
They stay the same, I think the MANE for these genes are all canonical too, to be clear, I changed the code.

Reply:
I tested: [66bacf5b23fc4272a98ce3c824cd90a0](https://console.cloud.google.com/dataproc/jobs/66bacf5b23fc4272a98ce3c824cd90a0?region=us-central1&project=broad-mpg-gnomad) and [dbb12f52789441c2a30445eb86aba2dc](https://console.cloud.google.com/dataproc/jobs/dbb12f52789441c2a30445eb86aba2dc?region=us-central1&project=broad-mpg-gnomad)